wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

```diff
@@ -0,0 +1,39 @@
+{% load wagtailadmin_tags i18n %}
+<template data-dialog>
+    <div
+        id="{{ id }}"
+        aria-labelledby="title-{{ id }}"
+        aria-hidden="true"
+        class="{{ class_name }} w-dialog"
+    >
+        <div data-a11y-dialog-hide class="w-dialog__overlay"></div>
+        <div class="w-dialog__box">
+            <button type="button" data-a11y-dialog-hide aria-label="{% trans 'Close dialog' %}" class="w-dialog__close-button">
+                {% icon name='cross' class_name="w-dialog__close-icon" %}
+            </button>
+
+            {% if message_heading and message_icon_name %}
+                <div class="w-dialog__message w-dialog__message--{{ message_status }}">
+                    {% icon name=message_icon_name class_name="w-dialog__message-icon" %}
+                    <div class="w-dialog__message-header">
+                        <strong class="w-dialog__message-heading">{{ message_heading }}</strong>
+                        {% if message_description %}<p class="w-dialog__message-description ">{{ message_description }}</p>{% endif %}
+                    </div>
+                </div>
+            {% endif %}
+
+            <div class="w-dialog__content">
+                <h2 class="w-dialog__title w-h1" id="title-{{ id }}">
+                    {% if icon_name %}{% icon name=icon_name class_name="w-dialog__icon" %}{% endif %}
+                    {{ title }}
+                </h2>
+
+                {% if subtitle %}
+                    <p class="w-dialog__subtitle w-help-text">{{ subtitle }}</p>
+                {% endif %}
+
+
+                {% comment %}
+    This markup is intentionally left without closing div tags so that the contents can be populated with child elements between dialog and enddialog
+    For the end tags please see end-dialog.html
+                {% endcomment %}
```

Comment:
Just wondering. If we put the closing divs all in one line within the comment - will curlylint be tricked into thinking the Dom structure is ok? Or maybe within a html comment. 

Or even within an if statement that will never be true?

Pretty hacky but just a thought.

Reply:
Ok. Crazy idea. We use one template only and provide a context variable for is_start or similar and then put the whole html in one file?